From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RobotsNeverDie <mykehype@gmail.com>
Date: Sat, 17 Oct 2026 03:07:33 -0400
Subject: [PATCH] 9933: Fix 64bit integer overflow in native sleigh compiler on
 Windows

Sleigh: Use std::stoull in scan_number to fix 64bit integer
 overflow on Windows
---
 Ghidra/Features/Decompiler/src/decompile/cpp/slghscan.cc | 2 +-
 Ghidra/Features/Decompiler/src/decompile/cpp/slghscan.l  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/slghscan.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/slghscan.cc
index 1ce00673ae..bdb2178b90 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/slghscan.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/slghscan.cc
@@ -1611,7 +1611,7 @@ int4 scan_number(char *numtext,SLEIGHSTYPE *lval,int4 radix,bool signednum)
 {
   uintb val;
   try {
-    val = std::stoul(numtext,(size_t *)0,radix);
+    val = std::stoull(numtext,(size_t *)0,radix);
   }
   catch(...) {
     return BADINTEGER;
diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/slghscan.l b/Ghidra/Features/Decompiler/src/decompile/cpp/slghscan.l
index b720e966c2..acc923e0c6 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/slghscan.l
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/slghscan.l
@@ -461,7 +461,7 @@ int4 scan_number(char *numtext,SLEIGHSTYPE *lval,int4 radix,bool signednum)
 {
   uintb val;
   try {
-    val = std::stoul(numtext,(size_t *)0,radix);
+    val = std::stoull(numtext,(size_t *)0,radix);
   }
   catch(...) {
     return BADINTEGER;
-- 
2.45.1

